src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -247,11 +247,22 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             }
             else if (parameter.ParameterType == typeof(string) || TryParseMethodCache.HasTryParseMethod(parameter))
             {
-                // We're in the fallback case and we have a parameter and route parameter match so don't fallback
-                // to query string in this case
-                if (factoryContext.RouteParameters is { } routeParams && routeParams.Contains(parameter.Name, StringComparer.OrdinalIgnoreCase))
+                // 1. We only bind from route values only, if route paramters are non-null and the parameter name is in that set.
+                // 2. We only bind from query only, if route paramters are non-null and the parameter name is NOT in that set.
+                // 3. Otherwise, we fallback to route or query if route paramters is null (it means we don't know what route parameters are defined). This case only happens
```

Comment:
```suggestion
                // 3. Otherwise, we fallback to route or query if route parameters is null (it means we don't know what route parameters are defined). This case only happens
```